express.js

```diff
@@ -1,9 +1,76 @@
+const express = require('express')
+const path = require('node:path')
+const { PORT, ROUTES, REQUEST_METHODS } = require('./contants.js')
+
+const app = express()
+
+//* secio-note (1)
+app.disable('x-powered-by')
+
+//* secio-note (2)
+app.use(express.json())
+
+//* secio-note (3)
+// @midudev ¿Si hubiesen archivos que no son estáticos u otros ficheros dentro del path especificado también son accesibles a la hora de entrar a sus respectivos paths?
+app.use(express.static(path.join(__dirname, 'assets')))
+
+app.all(ROUTES.ROOT, (req, res) => {
+  if (req.method === REQUEST_METHODS.GET) {
+    return res.status(200).send('<h1>¡Hola mundo!</h1>')
+  }
+  return res.status(405).end()
+})
+app.all(ROUTES.LOGO, (req, res) => {
+  //* secio-note (4)
+  if (req.method === REQUEST_METHODS.GET) {
+    const logoPath = path.join('assets', 'logo.webp')
+    // @midudev es la primera vez que veo "sendFile" por lo que no se si sea necesario pasarle ese objeto con root o si simplemente es mejor enviar el path completo.
+    return res.status(200).sendFile(logoPath, { root: path.join(__dirname) })
+  }
+  return res.status(405).end()
+})
+
+// @midudev y ya para terminar (^^), no se si la forma que hice de validar y enviar asi el error 405 sea la mejor, por lo que me gustaria saber tu opión ya que tanto en el ejercicio de node:http como en el de express me quedo la duda si era optimo o recomendable hacerlo así, ya que hasta ahora nunca maneje este tipo de errores, así como tampoco utilice hasta ahora el app.all() y no me queda del todo claro como hacer este tipo de validaciones de una manera correcta, gracias por todo, pasa un lindo día.
```

Comment:
Sí, está bien que en cada ruta conocida estés devolviendo el 405 si el método no se acepta. Está bien. 👍 

405 significa "Esta ruta la conozco pero no soporto este método" y es normal que lo hagas por cada endpoint como lo has hecho.

Reply:
Vaya, casi me asusto, pensaba estaba repidiendo mucho código, y que había otra forma para simplicar el proceso al ser algo repitivo, ya sabes simpre esta esa espinita de "clean code" , "DRY", etc, y uno quiere hacerlo todo más corto, pero si me lo dice midu me quedo más tranquilo, de hecho nunca hice algo parecido así que apartir de ahora agregaré esa validación en cada ruta para crear APIs más robustas y semánticas 😊.
```suggestion
// @midudev y ya para terminar (^^), no se si la forma que hice de validar y enviar asi el error 405 sea la mejor, por lo que me gustaria saber tu opión ya que tanto en el ejercicio de node:http como en el de express me quedo la duda si era optimo o recomendable hacerlo así, ya que hasta ahora nunca maneje este tipo de errores, así como tampoco utilice hasta ahora el app.all() y no me queda del todo claro como hacer este tipo de validaciones de una manera correcta, gracias por todo, pasa un lindo día.
```